### PR TITLE
Escape ID in Util.getSelectorFromElement

### PR DESCRIPTION
**Pull request by Johann-S**
Related to #22060

### CODE REVIEW

File: js/src/util.js [hunk 1]
```diff
@@ -87,6 +87,10 @@ const Util = (($) => {
     }
   }
 
+  // Thanks to learn.jquery.com see : http://bit.ly/1Tyr4SZ
+  function escapeId(selector) {
```
Comment (Johann-S) [inline js/src/util.js]: Maybe we can avoid to create a new function for that :thinking:

**Comment by XhmikosR** [comment]
If we required v3 there's already a function.

I'll review when I'm back in an hour or so.

**Comment by Johann-S** [comment]
it seems it wasn't working in v3 too

**Comment by XhmikosR** [comment]
We didn't call the function. How do you expect it to work ?

**Comment by Johann-S** [comment]
Sorry I don't get it of which function do you talk about ?

**Comment by XhmikosR** [comment]
https://api.jquery.com/jQuery.escapeSelector/

**Review by XhmikosR — APPROVED** [review]

**Comment by Johann-S** [comment]
`$.escapeSelector` don't produce the same result for example : 
```html
<div id="test.id"></div>
```
```js
$($.escapeSelector('#test.id')).length // => 0
$('#test.id'.replace(/(:|\.|\[|\]|,|=|@)/g, '\\$1')).length // => 1
```

**Comment by Johann-S** [comment]
Finally I found a way to use `$.escapeSelector` :smile:

**Comment by XhmikosR** [comment]
We could add a comment about `escapeSelector`. Or we could use that if available maybe?

**Comment by Johann-S** [comment]
I added a comment :+1:

File: js/src/util.js [hunk 2]
```diff
@@ -112,6 +111,12 @@ const Util = (($) => {
         selector = element.getAttribute('href') || ''
       }
 
+      // if it's an ID
+      if (selector.charAt(0) === '#') {
+        // we escape IDs in case of special selectors (selector = '#myId:something')
+        selector = $.escapeSelector(selector).substr(1)
```
Comment (XhmikosR) [inline js/src/util.js]: This will fail if the user uses jQuery < 3.

Comment (Johann-S) [reply js/src/util.js]: Nice catch ! :+1:

Comment (XhmikosR) [reply js/src/util.js]: Just keep our function like in your first patch, add a comment that this is required for jQuery < 3 compatibility and call it conditionally maybe if it's defined.

Comment (Johann-S) [reply js/src/util.js]: done :+1:

**Pull request merged**